ios/Handlers/RNTapHandler.m

```diff
@@ -81,14 +99,35 @@ - (void)touchesMoved:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
         }
     }
 
+    if ([self shouldFailUnderCustomCriteria]) {
+        self.state = UIGestureRecognizerStateFailed;
+        return;
+    }
+
     self.state = UIGestureRecognizerStatePossible;
     [self triggerAction];
 }
 
+- (BOOL)shouldFailUnderCustomCriteria
+{
+    CGPoint trans = [self translationInView:self.view];
+    if (TEST_MAX_IF_NOT_NAN(fabs(trans.x), _maxDeltaX)) {
+        return YES;
+    }
+    if (TEST_MAX_IF_NOT_NAN(fabs(trans.y), _maxDeltaY)) {
+        return YES;
+    }
+    if (TEST_MAX_IF_NOT_NAN(fabs(trans.y * trans.y + trans.x + trans.x), _maxDistSq)) {
+        return YES;
+    }
+    return NO;
+}
+
 - (void)touchesEnded:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
 {
-    [super touchesEnded:touches withEvent:event];
-    if (_numberOfTaps == _tapsSoFar) {
+    if (_numberOfTaps == _tapsSoFar && self.numberOfTouches >= _minPointers) {
```

Comment:
From my understanding `self.numberOfTouches` refers to the current number of pointers placed on the screen. This is inconsistent with Android implementation where we take maximum number of touches over the duration of gesture.